src/client/src/core/util/validation/__tests__/rules.spec.ts

```diff
@@ -1057,3 +1059,30 @@ describe('core/util/validation/rules::validVideoURL', () => {
     );
   });
 });
+
+describe('core/util/validation/rules::digitsOnlyWithTwoDecimalPoints', () => {
+  test('digitsOnlyWithTwoDecimalPoints:: with empty string', () => {
+    const result = digitsOnlyWithTwoDecimalPoints('');
+    expect(result).toEqual(true);
+  });
+
+  test('digitsOnlyWithTwoDecimalPoints:: with two decimal points', () => {
+    const result = digitsOnlyWithTwoDecimalPoints('123.98');
+    expect(result).toEqual(true);
+  });
+
+  test('digitsOnlyWithTwoDecimalPoints:: with one decimal points', () => {
+    const result = digitsOnlyWithTwoDecimalPoints('123.9');
+    expect(result).toEqual(true);
+  });
+
+  test('digitsOnlyWithTwoDecimalPoints:: with three decimal points', () => {
+    const result = digitsOnlyWithTwoDecimalPoints('123.998');
+    expect(result).toEqual('Should be a valid number (xxx.xx)');
+  });
+
+  test('digitsOnlyWithTwoDecimalPoints:: with no decimal points', () => {
+    const result = digitsOnlyWithTwoDecimalPoints('123.');
```

Comment:
this should fail? try `^\d+?(?:\.\d{1,2})?$`

Reply:
it's working. 123.  is taken as 123.00

Reply:
it works because you do transforming at `create` method `(Math.floor(this.expense.amount * 100) / 100).toFixed(2)`
But UI wise it looks wrong